src/main/java/apoc/map/Maps.java

```diff
@@ -178,18 +178,39 @@ public Object get(@Name("map") Map<String,Object> map, @Name("key") String key,
     }
 
     @UserFunction
-    @Description("apoc.map.removeKey(map,key)")
-    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key) {
+    @Description("apoc.map.removeKey(map,key,recursive:false) - remove the key from the map (recursively if recursive is true)")
+    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key, @Name(value="recursive", defaultValue = "false") boolean recursive) {
+        if (!map.containsKey(key)) {
+            return map;
+        }
         Map<String, Object> res = new LinkedHashMap<>(map);
         res.remove(key);
+        if (recursive) {
+            for (Map.Entry<String, Object> entry : res.entrySet()) {
+                if (entry.getValue() instanceof Map) {
```

Comment:
we should manage also collections, what if we have the following use-case:

```
RETURN apoc.map.removeKey({a:1,b:2,c:[{a:3,b:4}, {a:4,b:5}]},'a', true)
```

I expect that the result is:

```
{b:2,c:[{b:4}, {b:5}]}
```

Please add this as test-case